src/test/java/sysc4806/project24/mini_survey_monkey/integration/ControllerIntegrationTest.java

```diff
@@ -90,5 +90,43 @@ public void testValidRootRedirection() throws Exception {
                 .andExpect(status().is3xxRedirection())
                 .andExpect(header().string("Location", startsWith("/home")));
     }
+
+    /**
+     * Tests when a survey goes from draft to open
+     */
+    @Test
+    public void testOpeningSurvey() throws Exception {
+        // Create a survey
+        mockMvc.perform(post("/create"));
+        int surveyId = 1;
+
+        // /survey/id should initially be invalid
+        mockMvc.perform(get("/survey/" + surveyId))
+                .andExpect(status().is4xxClientError());
```

Comment:
When I run `mvn test`, this part of the test fails. Expects a client error but returns SUCCESSFUL.

```
[ERROR] Failures: 
[ERROR]   ControllerIntegrationTest.testOpeningSurvey:105 Range for response status value 200 expected:<CLIENT_ERROR> but was:<SUCCESSFUL>
```

Reply:
I'm able to recreate this in the browser too.

When I create a `DRAFT` survey. I am able to go to `~/survey/1` - even though it is a `DRAFT`.

Reply:
In the browser, I am also able to go the `/create/id` page for a survey that is `OPEN`.

I suspect is is something to do with the interceptor not actually intercepting.